### PR TITLE
API v1: move data sources endpoints under `/vaults/{vId}`

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Move all v1 data_sources endpoints under `vaults/{vId}` with backward compatibility
- Document the new endpoints

Potentially easier to review commit by commit. All files were copied first. For each `vaults/{vId}` files I did the following:
- updated the docstring to add vId to the path
- add the backward compatibility VaultResource retrieval code
- changed the `if` condition right after finding the dataSource to check vault sId matching

Important to review are the imports in the legacy endpoints which I had to write.

Dev test:
- [x] legacy document upsertion works
- [x] legacy document deletion works
- [x] legacy csv upload works
- [x] legacy search works

## Risk

High: could break users and connectors if erroneous

## Deploy Plan

- deploy `front`

### CODE REVIEW

**Comment by spolu** [comment]
Tests run in development:
```
# curl http://localhost:3000/api/v1/w/mkk5BKM98V/data_sources/foo3/documents/foo122 \
#     -H "Authorization: Bearer sk-bf81223a0c27fd484b5b0e8aabf33565" \
#     -H "Content-Type: application/json" \
#     -d '{
#        "text": "foo2"
#     }'

# curl -XDELETE http://localhost:3000/api/v1/w/mkk5BKM98V/data_sources/foo3/documents/foo122 \
#     -H "Authorization: Bearer sk-bf81223a0c27fd484b5b0e8aabf33565"


# curl -XPOST http://localhost:3000/api/v1/w/mkk5BKM98V/data_sources/foo3/tables/csv \
#     -H "Authorization: Bearer sk-bf81223a0c27fd484b5b0e8aabf33565" \
#     -H "Content-Type: application/json" \
#     -d '{
#       "name": "footable",
#       "truncate": true,
#       "description": "desc",
#       "csv": "foo,bar,viz\n1,2,3\n4,5,6",
#       "tableId": "fooTable-1"
#     }'


# curl -XGET "http://localhost:3000/api/v1/w/mkk5BKM98V/data_sources/foo3/search?query=foo&top_k=3&full_text=false" \
#     -H "Authorization: Bearer sk-bf81223a0c27fd484b5b0e8aabf33565"
```

File: front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/index.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you're trying to access was not found",
+      },
+    });
+  }
+
+  const dataSources = await DataSourceResource.listByVault(auth, vault);
```
Comment (tdraier) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/index.ts:79]: we're fine that by default the legacy endpoint won't return managed datasources anymore (only website and folders) ?

File: front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/[dsId]/tokenize.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    vId = (await VaultResource.fetchWorkspaceGlobalVault(auth)).sId;
+  }
+
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
+    { origin: "v1_data_sources_tokenize" }
+  );
+  if (!dataSource || dataSource.vault.sId !== vId) {
```
Comment (tdraier) [inline front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/[dsId]/tokenize.ts:64]: Same issue here - legacy endpoint won't work for managed ds as it assumes global vault and managed ds are in system vault - all legacy endpoints used by connectors will break ?

Comment (spolu) [reply front/pages/api/v1/w/[wId]/vaults/[vId]/data_sources/[dsId]/tokenize.ts:64]: Yes realised that while commuting back. Will adress tomorrow.

There's a bit more work to allow searching connections as well.

**Review by tdraier — APPROVED** [review]

**Comment by spolu** [comment]
Tests to run locally: add a connection and ingest (notion with at least one database)

**Comment by spolu** [comment]
run a full ingestion of Notion without issue locally. Merging

**Pull request merged**